.ado/publish.yml

```diff
@@ -0,0 +1,48 @@
+# Publishing VSCode Extensions https://code.visualstudio.com/api/working-with-extensions/publishing-extension
```

Comment:
Is the name 'publish.yml' required? We're probably going to need publish ADO pipelines for other artifacts too (e.g. pip and npm packages), so the name might want to be more specific if this will be just for the VS Code extension.

Reply:
This .yml is planned for publishing everything (under the same version). And actually, the original sketch contains building the Py packages too, but Py publishing was waiting for the ESRP team approval/reply, so for now I decided to merge the VSCE part only.

After I pass through the whole procedure with ESRP, I'll add Py packages to this file.